### PR TITLE
Fix Data-View-Translator and spec test

### PR DESCRIPTION
**Pull request by oehm-smith**
* Tests had false positive
* Updated translator to use the one common method

### CODE REVIEW

File: src/client/app/shared/json-data-view-model/data-view-translator.ts [hunk 1]
```diff
@@ -35,16 +26,7 @@ export class DataViewTranslatorService {
...
-      let viewTypedPointer: TypedPointer = fieldMap.viewTypedPointer;
-      let viewValue: string = JsonPointerService.get(viewModel, viewTypedPointer.pointer);
-      if (dataTypedPointer.type === 'number') {
-        JsonPointerService.set(dataModel, dataTypedPointer.pointer, viewValue !== null ? parseFloat(viewValue) : null);
-      } else {
-        JsonPointerService.set(dataModel, dataTypedPointer.pointer, viewValue);
-      }
-    }
+      DataViewTranslatorService.translate(viewModel, dataModel, fieldMappings, true);
   }
```
Comment (lbodor) [inline src/client/app/shared/json-data-view-model/data-view-translator.ts]: It's great to now have a single translate function. :+1:

File: src/client/app/shared/json-data-view-model/data-view-translator.ts [hunk 2]
```diff
@@ -71,9 +53,62 @@ export class DataViewTranslatorService {
...
+            + DataViewTranslatorService.padTwo(date.getMinutes()) + ':'
+            + DataViewTranslatorService.padTwo(date.getSeconds());
+
+        let dateTime: string = dateStr + 'T' + timeStr + '.000Z';
+
+        console.debug(`  DataViewTranslatorService - getFormatDateToDatetimeString - return '${dateTime}'`);
+        return dateTime;
+    }
+
+    private static isaDate(date: Date): boolean {
```
Comment (lbodor) [inline src/client/app/shared/json-data-view-model/data-view-translator.ts]: Rename to isDate please. Keep articles out of identifiers.

**Comment by oehm-smith** [comment]
I fixed lint problems and removed code from commit that wasn't intended (just yet).

File: src/client/app/shared/json-data-view-model/data-view-translator.ts [hunk 3]
```diff
@@ -71,9 +53,62 @@ export class DataViewTranslatorService {
...
+        if (Object.prototype.toString.call(date) === "[object Date]") {
+            if (isNaN(date.getTime())) {  // d.valueOf() could also work
+                return false;
+            }
+            else {
+                return true;
+            }
```
Comment (lbodor) [inline src/client/app/shared/json-data-view-model/data-view-translator.ts]: Replace 6 lines above with `return !isNaN(date.getTime())`

File: src/client/app/shared/json-data-view-model/data-view-translator.ts [hunk 4]
```diff
@@ -71,9 +53,62 @@ export class DataViewTranslatorService {
...
+     * @return formatted string or itself if a string to begin with
+     */
+    private static formatDate(date: any): string {
+        if (!DataViewTranslatorService.isaDate(date)) {
+            return <string> date;
+        }
+        return DataViewTranslatorService.getFormatDateToDatetimeString(<Date> date);
+    }
+
+    private static getFormatDateToDatetimeString(date: Date): string {
```
Comment (lbodor) [inline src/client/app/shared/json-data-view-model/data-view-translator.ts]: Bad name, this method is not a getter. A better name is `formatDatetime`

**Pull request merged**